service/devdoc/device_method_requirements.md

```diff
@@ -65,8 +65,30 @@ Request-Id: <guid>
   "payload": <DeviceMethod.params.payload>
 }
 ```
-**]**  
+**]**
 
-**SRS_NODE_IOTHUB_DEVICE_METHOD_16_009: [** The `invokeOn` method shall invoke the `done` callback with a standard javascript `Error` object if the method execution failed. **]**  
+**SRS_NODE_IOTHUB_DEVICE_METHOD_16_009: [** The `invokeOn` method shall invoke the `done` callback with a standard javascript `Error` object if the method execution failed. **]**
 
-**SRS_NODE_IOTHUB_DEVICE_METHOD_16_010: [** The `invokeOn` method shall invoke the `done` callback with a `null` first argument, a result second argument and a transport-specific response third argument if the method execution succeeded. **]**
\ No newline at end of file
+**SRS_NODE_IOTHUB_DEVICE_METHOD_16_010: [** The `invokeOn` method shall invoke the `done` callback with a `null` first argument, a result second argument and a transport-specific response third argument if the method execution succeeded. **]**
+
+### invokeOnModule(deviceId: string, moduleId: string, done: DeviceMethod.ResponseCallback): void;
+
+**SRS_NODE_IOTHUB_DEVICE_METHOD_18_001: [** The `invokeOnModule` method shall throw a `ReferenceError` if `deviceId` or `moduleId` is falsy. **]**
+
+**SRS_NODE_IOTHUB_DEVICE_METHOD_18_002: [** The `invokeOnModule` method shall construct an HTTP request using information supplied by the caller, as follows:
+```
+POST /twins/<deviceId>/modules/<moduleId>/methods?api-version=<version> HTTP/1.1
+Authorization: <config.sharedAccessSignature>
+Content-Type: application/json; charset=utf-8
+Request-Id: <guid>
+{
+  "methodName": <DeviceMethod.params.methodName>,
+  "timeoutInSeconds": <DeviceMethod.params.timeoutInSeconds>,
```

Comment:
>timeoutInSeconds [](start = 3, length = 16)

don't we have a more complicated timeout structure here? why would it be different from the responseTimeoutInSeconds/connectTimeoutInSeconds properties that we have for invokeOn?